new pull request

rareaquaticbadger opened the pull request:
Made a few changes and worked a little bit on vignette.  Will work more on it, but probably not until after christmas.  Not sure how to run R CMD check (how do I find the contiBAIT package?), but hopefully this will still work ok.  Sweave is working a treat on the vignette.

I've added a function splitting at gap locations.  It's one of 2-3 additional functions I've been working on.  Hopefully it won't be too much of a headache to incorporate.

rareaquaticbadger committed:
more vignette stuff

rareaquaticbadger committed:
- added more to vignette
- added check to plotContigOrder to check that all elements of LG are from the same chromosome before using start positions to order
- found bug in combineZeroDists
- reverted orderAllLinkageGroups as orderedGroups was being overwritten with latest code so only last LG was returning.
- Fixed error where if only 1 group in LG, best_order wasn't being returned from orderContigsGreedy
- combineZeroDists now checks that there are no empty libraries or contigs.

Merge remote-tracking branch 'upstream/devel'

Conflicts:
	R/orderAllLinkageGroups.R
	vignettes/contiBAIT.aux
	vignettes/contiBAIT.log
	vignettes/contiBAIT.out
	vignettes/contiBAIT.toc

oneillkza commented:
To run R CMD check, jump up one directory from where you have the ContiBAIT code, and run:

``` BASH
$ R CMD build ContiBAIT
$ R CMD check contiBAIT_0.99.0.tar.gz
```

Ensure both run without any errors or warnings. `R CMD check` also produces a report in `contiBAIT.Rcheck/` which you can refer to regarding the errors.

(Note I just tried this on your code and the build failed.)

rareaquaticbadger committed:
Argument change in function not changed in vignette

rareaquaticbadger committed:
few bug fixes found with R CMD check

rareaquaticbadger committed:
added dependency

rareaquaticbadger committed:
bug fix

rareaquaticbadger commented:
Thanks,

I've fixed the bug (changed an argument name, but forgot to change it in the vignette). Try now and it should work.

Minimal check errors with R CMD check. The only notes I get are:
- checking dependencies in R code ... NOTE
  Package in Depends field not imported from: ‘BH’
  These packages need to be imported from (in the NAMESPACE file)
  for when this namespace is loaded but not attached.
  package 'methods' is used but not declared

(not sure which function is calling BH)
and:
- checking R code for possible problems ... NOTE
  strandSeqFreqTable : overlapStrand: no visible binding for global
  variable ‘rname’
  strandSeqFreqTable : overlapStrand: no visible binding for global
  variable ‘pos’

There's also mismatch documentation warning for mapGapFromOverlap that I'm working on.

Just a quick note: I'm away until Boxing day.  Will have email, but no computer to code on.  Can address issues after I'm back.

oneillkza commented:
Thanks Mark. I hope you're having a great Christmas break!

For when you come back:

Yeah, `NOTEs` from `R CMD check` aren't too much to worry about. The main thing is to avoid any `ERRORs` or `WARNINGs`. For reference, `BH` contains C++ headers from the Boost library that my distance calculation code uses, but it never gets called from within R.

The vignette looks pretty good, though I do notice an "(IN DEVEL)" line in it. 

I also noticed you added `mapGapFromOverlap` as new functionality. I haven't tested it yet, and it doesn't come with an example (right now its example just points to the one for `strandSeqFreq`), so I'm a little leery of merging it. 

One comment I have on the code itself is that BioConductor recommends using native BioConductor classes for representing data, so rather than having `sceFile` and `gapFile` be data.frames (which promptly get converted to `GRanges` objects), it would be better if they were passed in as `GRanges` objects from the start. For the example, this can be achieved fairly easily using the `import.bed` function from `rtracklayer`.

Also for the example, we should provide the corresponding BED files. This would probably involve moving the BAM files currently in `inst/extdata` to a separate `inst/extdata/bam` directory (and updating the example for `strandSeqFreq` to look in that directory), then creating a separate `inst/extdata/bed` directory.

I could have a go at this, but I don't have the strand state change BED file to use for the example.

oneillkza commented on R/mapGapFromOverlap.R:
```diff
@@ -0,0 +1,71 @@
+####################################################################################################
+#' mapGapFromOverlap -- function to co-localize strand state changes with assembly gaps 
+#' 
+#' @param sceFile data.frame of strand state change locations in BED format 
```
These should be `GRanges` objects, not `data.frame`

oneillkza commented on R/mapGapFromOverlap.R:
```diff
@@ -0,0 +1,71 @@
+#' @import GenomicRanges
+#' @import DNAcopy 
+#' @importFrom S4Vectors DataFrame
+#' @example inst/examples/strandSeqFreqTable.R
```
Still needs an example!

oneillkza commented on R/mapGapFromOverlap.R:
```diff
@@ -0,0 +1,71 @@
+#' @param overlapNum  Minimal number of strand state changes that overlap with a gap before assembly is cut at that location
+#' @param verbose prints messages to the terminal (default is TRUE)
+#' 
+#' @return a list containing two matrices: a StrandFreqMatrix of W:C read frequencies, and a StrandReadMatrix of read counts
```
Pretty sure this is not what it's returning.

rareaquaticbadger committed:
-Multiple bug fixes
-Reworked example bam files to standard format
-Added parameter in strandSeqFreq to allow for BAIT output in future
-Updated mapGapFromOverlap code and manual
-Added mapGapFromOverlap example
-Worked on the vignette. Almost complete

rareaquaticbadger commented:
Hello,

Please see additions to the pull request.  It runs clean on R CMD check.

The biggest change: All of the example bam files are updated.  Having them assembled to a 1Mb split genome is weird. I don't do that any more (I split using strandSeqFreqTable), it's not standard, it's confusing to talk about in the vignette, and it means I can't provide an example for mapGap (because downloading from UCSC is pointless if the build is different). As such I've gotten the same regions, but aligned to GRCh38/hg38.  Please see the contiBAIT.R example to see how this now runs.  It looks the same as before.

I've added an example to the mapGapFromOverlap function.  Note this looks messy on the vignette as the URL contained in the rtracklayer command doesn't wrap around, and I don't know how to do this in sweave.

The vignette is mostly done.  Need a few other bits and bobs to finish off.

I've also added a parameter for strandSeqFreqTable to output just the W and C read numbers.  I plan to use this to port BAIT over to an entirely R-only package. The final output of contiBAIT should be BAIT ideograms anyway to show that the contigs now behave like full chromosomes, so I think it's worth including.

Cheers.

oneillkza commented:
Thanks Mark! It's looking a lot better now.

I've checked your branch, and it does indeed build and pass check. Do you think you could resolve the upstream conflicts? It looks like you didn't update from the work I did last week.

I'm going away for a few days, but will be back on Sunday, so should have a chance to look at it in more detail then. Next week I'll also be able to look at fixing the BED output (issue https://github.com/oneillkza/ContiBAIT/issues/16 )

rareaquaticbadger committed:
-merged upstream devel
-resolved conflicts with plotWCDistribution vs plotWDDistribution
-added split chromosome data to examples

Merge remote-tracking branch 'upstream/devel'

Conflicts:
	DESCRIPTION
	R/barplotLinkageGroupCalls.R
	R/contiBAIT.R
	man/barplotLinkageGroupCalls-LinkageGroupList-method.Rd

rareaquaticbadger commented:
See changes.  Passed R CMD check, and looks good.

Few bug issues, but nothing major.  If I get the chance, I'll have a look at the BED output.  In the processes of making a very simple BAIT plotter (if we feed the output BED file back into strandSeqFreqTable we'll be able to create BAIT ideograms to check the conformity of the ordering). I still have a few other functions I'm working on, but they're not integral to the core functionality.

The other issue I would recommend looking at is combining clusterContigs, reorientLinkageGroups and reorientStrandTable together.  If this is possible, we should also just cut exampleStrandStateMatrix[[2]], as this is only used in the first two of these functions, and is a pretty weird data table to return for the most part.  I was going to look at porting over the code from preprocessStrandTable to clusterContigs (I think clusterContigs should be natively removing the WC calls, if just to prevent errors from reading in the wrong strandStateMatrix).

I also really think that reorientLinkageGroups improvements should be a priority (we talked about this last time we talked), as it's definitely sub optimal right now.

rareaquaticbadger committed:
Major change: Removed the strandStateMatrix in which WC calls are removed. Advantages:
1. clusterContigs fails if strandStateMatrix is run with WC values, so makes sense to remove them within this function
2. This dataframe was confusing as output from preprocessStrandTable
3. only clusterContigs uses this
Vignette and examples updated to reflect this change.
Tried to get reorientLinkageGroups working a bit better:
- now reads in all strands (WW, WC & CC), but substitutes WC for NAs after consensus computed.
- No long uses '2' as intermediate when flipping '3' values to '1' values so as not to overwrite WC calls.

rareaquaticbadger commented:
See changes. Clean R CMD check.  I've gotten rid of strandStateMatrix[[2]], and moved the WC removal line to cluster contigs. Updated manuals, examples and vignette accordingly.
Tried to get reorientLinkageGroups working, but it's still acting weird.
